src/sql/parts/notebook/cellViews/code.component.ts

```diff
@@ -135,6 +136,23 @@ export class CodeComponent extends AngularDisposable implements OnInit, OnChange
 		}
 	}
 
+	private updateConnectionState(isConnected: boolean) {
+		if (this.isSqlCodeCell()) {
+			let cellUri = this.cellModel.cellUri.toString();
+			if (!isConnected && this._model.notebookOptions.connectionService.isConnected(cellUri)) {
```

Comment:
this._model.notebookOptions.connectionService is always a valid object? shouldn't we guard it ?

Reply:
Definitely, will fix